### PR TITLE
User Management functionality

### PR DESCRIPTION
**Pull request by Outragedpillow**
Features:
Added functionality to the reset user password button. This gives the user a temporary password to use and will have them reset it at the next login
uiPwResetInit

Success: 
uiResetPwSuccess

If an account is an admin give an error:
uiResetPwFail

Also added functionality to delete user button. For the moment allows deletion of both admin and student user types.
success: 
uiDeleteSuccess

These use a toast to display success or error that auto closes after a set time (5 seconds).
Rather than have a separate toast for different situations, made a dynamic toast that accepts a message, type, and has a bool for whether it should be displayed or not. 

Fixes:
#127 This was also fixed in my last PR and that code is implemented here as well
#124 modal now closes on submit. Response object was being misused.
#108 fixed issue. Enum validation was not working and had to be done manually for now. See Http/Controllers/v1/UserController.php

### CODE REVIEW

File: resources/js/Pages/Users.tsx [hunk 1]
```diff
@@ -212,61 +278,93 @@ export default function Users({ auth }: PageProps) {
...
                     </h3>
+                    <h4 className="font-bold text-l text-secondary">
+                        <br />
+                        <p>Note: Only for non-administrator accounts.</p>
+                    </h4>
                     <p className="py-4"></p>
                     <div className="modal-action">
-                        <button className="btn btn-warning">Reset</button>
+                        <button
+                            className="btn btn-warning"
```
Comment (calisio) [inline resources/js/Pages/Users.tsx:290]: For consistency sake (until we get a component that is used for all modals), could you modify these modals to follow the same format as left menu management modals? ie. color should be btn-error (or you can change the left menu modals to btn-warning... either way), close button should be left aligned, and all modals should have an "x" on the top right corner. Super small changes (and I know they existed before your PR) but keeps the UI consistent!

File: resources/js/Components/forms/AddUserForm.tsx [hunk 2]
```diff
@@ -25,99 +30,119 @@ export default function AddUserForm({ onSuccess }: { onSuccess: () => void }) {
...
+                    </div>
+                    <input
+                        type="text"
+                        className="input input-bordered w-full"
+                        {...register("username", {
+                            required: "Username is required",
+                            maxLength: {
+                                value: 50,
+                                message:
+                                    "Username shoudl be 50 characters or less",
```
Comment (calisio) [inline resources/js/Components/forms/AddUserForm.tsx:114]: small type on "should"

File: resources/js/Pages/Users.tsx [hunk 3]
```diff
@@ -212,61 +278,93 @@ export default function Users({ auth }: PageProps) {
...
-                    <CheckCircleIcon className="h-6" />
-                    <span>User deleted!</span>
+            {toast.isVisible && (
+                <div
+                    className={`toast transition-opacity duration-500 ease-out ${
+                        toast.isVisible ? "opacity-100" : "opacity-0"
+                    }`}
+                >
+                    <div className={`alert alert-${toast.type}`}>
+                        <CheckCircleIcon className="h-6" />
```
Comment (calisio) [inline resources/js/Pages/Users.tsx:363]: When its an error, can you render `<ExclamationCircleIcon className="h-6" />` instead?

**Review by calisio — CHANGES_REQUESTED** [review]
Super small changes! @PThorpe92 will review StudentNewPasswordController and UserController.

**Review by PThorpe92 — APPROVED** [review]
Other than what @calisio pointed out, looks good to me on the back-end. 

That UserRole enum issue has been driving me insane. I tried __everything__ and the only way to fix it was manually editing the user model like you did. :+1:

**Comment by Outragedpillow** [comment]
Thanks for the feedback @calisio! I will make these changes today. And yeah @PThorpe92 I wasn't sure if that was the best fix but it works for now

**Review by calisio — APPROVED** [review]

**Pull request merged**